### PR TITLE
Add SQL Datastore option to Bootstrap Node

### PR DESCRIPTION
**Pull request by albrow**
This PR is the same as https://github.com/0xProject/0x-mesh/pull/486 but rebased off of the `development` branch.

### CODE REVIEW

File: cmd/mesh-bootstrap/main.go [hunk 1]
```diff
@@ -66,8 +71,35 @@ type Config struct {
 	// P2PAdvertiseAddrs is a comma separated list of libp2p multiaddresses which the
 	// bootstrap node will advertise to peers.
 	P2PAdvertiseAddrs string `envvar:"P2P_ADVERTISE_ADDRS"`
-	// DataDir is the directory used for storing data.
+	// DataStoreType is the data store which will be used to store DHT data
```
Comment (albrow) [inline cmd/mesh-bootstrap/main.go]: Looking at these env vars again, I think it is confusing that they aren't separated in any way. Can we prefix the ones related to Postgres/SQL with `POSTGRES` or `SQL` and the ones related to LevelDB with `LEVEL
DB`?

For example `LEVELDB_DATA_DIR` and `SQL_DB_CONNECTION_STRING`.

Comment (opaolini) [reply cmd/mesh-bootstrap/main.go]: That makes sense, renamed.

**Comment by albrow** [comment]
LGTM! Let's just make sure CI passes.

File: integration-tests/integration_test.go [hunk 2]
```diff
@@ -308,7 +308,7 @@ func startStandaloneNode(t *testing.T, ctx context.Context, logMessages chan<- s
 	cmd.Env = append(
 		os.Environ(),
 		"VERBOSITY=5",
-		"DATA_DIR="+standaloneDataDir,
+		"LEVELDB_DATA_DIR="+standaloneDataDir,
```
Comment (albrow) [inline integration-tests/integration_test.go]: For `mesh` (not `mesh-bootstrap`) the environment variable is still called `DATA_DIR`.

**Pull request merged**